rfcs/0051-stop-transaction-processing.md

```diff
@@ -0,0 +1,146 @@
+different components of the node, namely, the CLI, the transaction pool, the
+snark pool, and the block producer.
+and the CLI. The procedure will be as follows:
+
+* The CLI will allow a slot at which the node will stop processing transactions
+  to be set. This will be done by setting a new optional flag `--slot-tx-end`
+  that will be used by the transaction pool, snark pool, and the block producer
+  to stop processing transactions after a certain slot. If the flag is not set,
+  the node will continue processing transactions as usual.
+* The node will stop accepting new transactions from the client after the slot
+  set by the CLI flag.
+* The transaction pool will stop accepting new transactions after the slot set
+  by the CLI flag.
+* The transaction pool will drop all transactions from the pool after the slot
+  set by the CLI flag.
+* The snark pool will stop accepting new snark works after the slot set by the
+  CLI flag.
+* The snark pool will drop all snark works from the pool after the slot set by
+  the CLI flag.
+* The block producer will stop including transactions in blocks after the slot
+  set by the CLI flag.
+
+Each of these procedures will be described in detail in the following sections.
+
+### Client submits transaction
+
+When a client sends a transaction to the node, the node will check if the
+current global slot is less than the slot set by the CLI flag `--slot-tx-end`.
+If the transaction is valid and the slot is less than the slot set by the CLI
+flag, the transaction will be added to the transaction pool. If the current
+global slot is equal or greater than the slot set by the CLI flag, the
+transaction will be rejected. The client will be notified of the rejection and
+the reason why the transaction was rejected.
+
+### Transaction pool
+
+When a transaction is received by the transaction pool from any source, the
+transaction pool will check if the current global slot is less than the slot set
+by the CLI flag. If the transaction is valid and the slot is less than the slot
+set by the CLI flag, the transaction will be added to the transaction pool. If
+the current global slot is equal or greater than the slot set by the CLI flag,
+the transaction will be rejected. The transaction pool will notify the source of
+the transaction of the rejection and the reason why the transaction was
+rejected.
+
+When the transaction pool handles the transition frontier, it will check if the
+current global slot is less than the slot set by the CLI flag. If the current
+global slot is equal or greater than the slot set by the CLI flag, the pool will
+drop all transactions. This includes transactions that were added to the pool
+before the slot set by the CLI flag.
+
+### Snark pool
+
+For the snark pool, the procedure is analogous to the transaction pool but with
+snark works instead of transactions.
+
+### Block producer
+
+When the block producer is producing a block, it will check if the current slot
```

Comment:
Block validation also needs to *reject incoming blocks* too, otherwise transactions could be included by other block producers and defeat the whole point of this modification.

Reply:
Yes, that's correct. That information was missing at the time of your comment but it has been added now.